client/ayon_core/hosts/resolve/plugins/load/load_media.py

```diff
@@ -0,0 +1,365 @@
+        data = {
+            key: version_data.get(key, str(None)) for key in add_keys
+        }
+
+        # add variables related to version context
+        data.update({
+            "representation": str(representation["_id"]),
+            "version": version_name,
+            "colorspace": colorspace,
+        })
+
+        return data
+
+    @classmethod
+    def get_item_color(cls, representation) -> str:
+        """Return item color name.
+
+        Coloring depends on whether representation is the latest version.
+        """
+        # Compare version with last version
+        project_name = get_current_project_name()
+        version = get_version_by_id(
+            project_name,
+            representation["parent"],
+            fields=["name", "parent"]
+        )
+        last_version = get_last_version_by_subset_id(
+            project_name,
+            version["parent"],
+            fields=["name"]
+        ) or {}
+
+        # set clip colour
+        if version.get("name") == last_version.get("name"):
+            return cls.clip_color_last
+        else:
+            return cls.clip_color
+
+    def _set_metadata(self, media_pool_item, context: dict):
+        """Set Media Pool Item Clip Properties"""
+
+        # Set the timecode for the loaded clip if Resolve doesn't parse it
+        # correctly from the input. An image sequence will have timecode
+        # parsed from its frame range, we will want to preserve that.
+        # TODO: Setting the Start TC breaks existing clips on update
+        #   See: https://forum.blackmagicdesign.com/viewtopic.php?f=21&t=197327
+        #   Once that's resolved we should enable this
+        # start_tc = media_pool_item.GetClipProperty("Start TC")
+        # if start_tc == "00:00:00:00":
+        #     from openpype.pipeline.editorial import frames_to_timecode
+        #     # Assume no timecode was detected from the source media
+        #
+        #     fps = float(media_pool_item.GetClipProperty("FPS"))
+        #     handle_start = context["version"]["data"].get("handleStart", 0)
+        #     frame_start = context["version"]["data"].get("frameStart", 0)
+        #     frame_start_handle = frame_start - handle_start
+        #     timecode = frames_to_timecode(frame_start_handle, fps)
+        #
+        #     if timecode != start_tc:
+        #         media_pool_item.SetClipProperty("Start TC", timecode)
+
+        # Set more clip metadata based on the loaded clip's context
+        metadata = {
+            "Clip Name": "{asset[name]} {subset[name]} "
+                         "v{version[name]:03d} ({representation[name]})",
+            "Shot": "{asset[name]}",
+            "Take": "{subset[name]} v{version[name]:03d}",
+            "Comments": "{version[data][comment]}"
+        }
+        for clip_property, value in metadata.items():
+            media_pool_item.SetClipProperty(clip_property,
+                                            value.format_map(context))
```

Comment:
We really like this on our end - but this MIGHT be studio specific.

This metadata is required for the load media stuff to work. We might want to expose these in project settings with these as defaults but allowing others to customize/remove things they don't need.

So we can have a Set metadata on load `dict` type of settings entry so once could set e.g. "Shot" metadata to "{asset[name]}", etc.

Reply:
I really like this as well. It had been moved to settings.